beta/iam/guides/how-to-messaging-delegate.md

```diff
@@ -0,0 +1,210 @@
+
+### Register the delegate with MobileCore
+
+The `MobileCore` framework maintains an optional property that holds reference to the `MessagingDelegate`.
+
+{% tabs %}
+{% tab title="iOS" %}
+
+Swift
+
+```swift
+/// defined in MobileCore.swift
+@objc public static var messagingDelegate: MessagingDelegate?
+```
+
+Assuming that `InAppMessagingHandler` is a class that implements `MessagingDelegate`, execute the following code to set the delegate in `MobileCore`:
+
+Swift
+
+```swift
+let myMessagingDelegate = InAppMessagingHandler()   
+
+MobileCore.messagingDelegate = myMessagingDelegate
+```
+
+{% endtab %}
+
+{% tab title="Android" %}
+
+{% endtab %}
+{% endtabs %}
+
+### MessagingDelegate protocol
+
+{% tabs %}
+{% tab title="iOS" %}
+
+The `MessagingDelegate` protocol, which is implemented in the `AEPServices` framework, is defined below:
+
+Swift
+
+```swift
+/// UI Message delegate which is used to listen for current message lifecycle events
+@objc(AEPMessagingDelegate)
+public protocol MessagingDelegate {
+    /// Invoked when the any message is displayed
+    /// - Parameters:
+    ///     - message: UIMessaging message that is being displayed
+    @objc
+    func onShow(message: Showable)
+
+    /// Invoked when the any message is dismissed
+    /// - Parameters:
+    ///     - message: UIMessaging message that is being dismissed
+    @objc
+    func onDismiss(message: Showable)
+
+    /// Used to find whether messages should be shown or not
+    /// - Parameters:
+    ///     - message: UIMessaging message that is about to get displayed
+    /// - Returns: true if the message should be shown else false
+    @objc
+    func shouldShowMessage(message: Showable) -> Bool
+
+    /// Called when `message` loads a URL
+    /// - Parameters:
+    ///     - url: the `URL` being loaded by the `message`
+    ///     - message: the Message loading a `URL`
+    @objc
+    optional func urlLoaded(_ url: URL, byMessage message: Showable)
+}
+```
+
+{% endtab %}
+
+{% tab title="Android" %}
+
+{% endtab %}
+{% endtabs %}
+
+### Using the Showable object in the protocol methods
+
+{% tabs %}
+{% tab title="iOS" %}
+
+Each of the methods implemented in the `MessagingDelegate` will be passed a [`Showable`](https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/ui/Showable.swift) object.  In the AEPMessaging SDK, the class implementing `Showable` is [`FullscreenMessage`](https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift). A `FullscreenMessage` object is wrapped in the [`Message`](./../public-classes-enums.md) class, which is the primary way for the developer to interact with the message.
+
+To get a reference to the `Message` object:
+
+1. Convert the `Showable` message parameter to `FullscreenMessage`
+1. Access the `parent` variable (note that `parent` is variable defined in `FullscreenMessage+Message.swift`, an extension in the AEPMessaging framework)
```

Comment:
this is intentional.  markdown will always order the numbered list correctly.  we use all `1` values so that if a new item needs to be inserted in the middle of the list, you don't ever have to worry about updating the numbers.

Reply:
Sounds good, didn't know about this 👀